### PR TITLE
Feature #1386:  Display the last protocol for the samples listed in the project (from lineage/process_measurement).

### PR DESCRIPTION
**Pull request by nafiz1001**
https://206.12.92.46/issues/1386

### CODE REVIEW

**Comment by nafiz1001** [comment]
So for the #587 PR has served me well.

**Review by ckostiw — CHANGES_REQUESTED** [review]

**Comment by ckostiw** [comment]
If a sample has no associated individual then the Cohort column shows "Loading..." forever.

Screen Shot 2023-07-12 at 2 32 21 PM

**Comment by ckostiw** [comment]
I ran sample QC on some samples but the last protocol still shows "Extraction". Shouldn't it now display "Sample QC" since that was the most recent protocol?
Screen Shot 2023-07-12 at 2 40 28 PM

Screen Shot 2023-07-12 at 2 41 12 PM

**Review by ckostiw — CHANGES_REQUESTED** [review]
The projects associated samples table is not updated after a template is submitted. I ran Normalization on some samples, which generates new samples, and they did not appear in the projects associated samples table until I refreshed the page.

**Review by UlysseFG — COMMENTED** [review]
This looks ok on the backend side ... not convinced it hsould go in protocol but nowhere seem that great so might as well go with this.

**Review by ckostiw — CHANGES_REQUESTED** [review]
Two tiny nitpicks

File: frontend/src/components/projects/ProjectsAssociatedSamples.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                return acc
+            }, {} as typeof lastProtocolBySampleID))
+        })()
+    }, [dispatch, sampleIDs])
+
+    const LastProtocol = useCallback(({ sampleID }: { sampleID: Sample['id'] }) => {
+        if (sampleID in lastProtocolBySampleID) {
+            return <>{lastProtocolBySampleID[sampleID]}</>
+        } else {
+            return <>-</>
```
[the remark this answers is not in the document]
Comment (nafiz1001) [reply frontend/src/components/projects/ProjectsAssociatedSamples.tsx]: Already pushed ;)

File: frontend/src/selectors.ts [hunk 2]
```diff
@@ -87,6 +88,7 @@ export const selectSampleKindsState = (state: RootState) => state.sampleKinds
 export const selectSampleKindsByID = (state: RootState) => state.sampleKinds.itemsByID as ItemsByID<SampleKind>
 export const selectSampleNextStepTemplateActions = (state: RootState) => state.sampleNextStepTemplateActions.items
 export const selectSampleTemplateActions = (state: RootState) => state.sampleTemplateActions
+export const selectProjectSamplesTable = (state: RootState) => state.projectSamplesTable as ProjectSamplesTable
```
[the remark this answers is not in the document]
Comment (nafiz1001) [reply frontend/src/selectors.ts]: i've sorted all of the selectors

**Review by ckostiw — APPROVED** [review]

**Comment by UlysseFG** [comment]
This works as intended ... We'll revisit using the lab feedback when it comes to the confusion of what "last protocol" means.

**Review by UlysseFG — APPROVED** [review]

**Pull request merged**